go/pkg/client/tree.go

```diff
@@ -378,19 +370,21 @@ func packageTree(t *treeNode, stats *TreeStats) (root digest.Digest, blobs map[d
 	}
 	sort.Slice(dir.Directories, func(i, j int) bool { return dir.Directories[i].Name < dir.Directories[j].Name })
 
-	for name, fn := range t.files {
-		dg := fn.ue.Digest
-		dir.Files = append(dir.Files, &repb.FileNode{Name: name, Digest: dg.ToProto(), IsExecutable: fn.isExecutable})
-		blobs[dg] = fn.ue
-		stats.InputFiles++
-		stats.TotalInputBytes += dg.Size
+	for name, n := range t.leaves {
+		if n.file != nil {
+			dg := n.file.ue.Digest
+			dir.Files = append(dir.Files, &repb.FileNode{Name: name, Digest: dg.ToProto(), IsExecutable: n.file.isExecutable})
+			blobs[dg] = n.file.ue
+			stats.InputFiles++
+			stats.TotalInputBytes += dg.Size
+		}
+		if n.symlink != nil {
```

Comment:
Can a node be both a file and a symlink? If not, that should be an error to avoid a tree with unexpected nodes.

However, the previous code gave precedence to the file. Maintaining that is also reasonable at this point.

Reply:
PTAL: Added an enforced consistency check to the internal fileSysNode type (should be exactly one of file/symlink/directory). Cannot unit test because it is an internal type.